src/config/parse.c

```diff
@@ -1411,6 +1411,22 @@ configParse(const Storage *storage, unsigned int argListSize, const char *argLis
                 if (!option.found)
                     THROW_FMT(OptionInvalidError, "invalid option '--%s'", arg);
 
+                // Handle boolean option with argument(only y/n allowed)
+                if (parseRuleOption[option.id].type == cfgOptTypeBoolean && optionArg != NULL)
+                {
+                    // Error on negated (--no-) options that also have an argument
+                    if (strncmp(strZ(optionName), OPTION_PREFIX_NEGATE, sizeof(OPTION_PREFIX_NEGATE) - 1) == 0)
+                        THROW_FMT(OptionInvalidError, "negated options cannot have an argument '--%s'", arg);
+
+                    // Validate argument/set negate when indicated
+                    if (strEqZ(optionArg, "n"))
+                        option.negate = true;
+                    else if (!strEqZ(optionArg, "y"))
```

Comment:
Since this statement breaks onto multiple lines it should be enclosed in braces, see https://github.com/pgbackrest/pgbackrest/blob/main/CODING.md#braces.